go-controller/pkg/ovn/ovn.go

```diff
@@ -225,84 +258,187 @@ func getPodNamespacedName(pod *kapi.Pod) string {
 	return util.GetLogicalPortName(pod.Namespace, pod.Name)
 }
 
+func NewControllerManager(ovnClient *util.OVNClientset, identity string, wf *factory.WatchFactory,
+	stopChan chan struct{}, libovsdbOvnNBClient libovsdbclient.Client, libovsdbOvnSBClient libovsdbclient.Client,
+	recorder record.EventRecorder, wg *sync.WaitGroup) *ControllerManager {
+	return &ControllerManager{
+		GenericController: GenericController{
+			client: ovnClient.KubeClient,
+			kube: &kube.Kube{
+				KClient:              ovnClient.KubeClient,
+				EIPClient:            ovnClient.EgressIPClient,
+				EgressFirewallClient: ovnClient.EgressFirewallClient,
+				CloudNetworkClient:   ovnClient.CloudNetworkClient,
+			},
+			watchFactory: wf,
+			recorder:     recorder,
+			nbClient:     libovsdbOvnNBClient,
+			sbClient:     libovsdbOvnSBClient,
+			podRecorder:  metrics.NewPodRecorder(),
+		},
+		defaultWg:       wg,
+		defaultStopChan: stopChan,
+		identity:        identity,
+	}
+}
+
+func (cm *ControllerManager) InitDefaultController(addressSetFactory addressset.AddressSetFactory) (*Controller, error) {
```

Comment:
It would be neat to initialize the default OvnController by throwing a default NAD to the ControllerManager work queue instead of having specific code.

Reply:
the ControllerManager work queue is the watcher handler queue of net-attach-def CRD.  I am not sure reuse that for the default controller is a good idea.

Reply:
I agree that (on paper ...) having a single flow is preferable; but, since we don't have the full picture available in this PR, there might be reasons to separate the flows.

@cathy-zhou is there any reason to have two different flow for initializing the controllers ? Is there anything wrong with @jcaamano 's suggestion of initializing all controllers as a response to queued events ?

Reply:
aside from the flow for create and initializing the default controller, all others are driven by event of net-attach-def creation/deletion, which is natural. I just didn't see the need to introduce another event driven controller. of course, if that is agreed on, I also don't see any problem with that approach. 

perhaps we can discuss this in the community meeting today.

Reply:
It would not be a different controller, it would be the same controller and the same code that is used to handle all the other controllers, so less code overall. If we are building a default Nad for the default controller and we have code to handle Nad, it also seems natural to not have more code for this. But indeed these depends on how the controller manager operates with the other controllers which we are not seeing much in this PR.

Reply:
As I said, this would be neatness.

But the discussion of having a level driven controller instead of a event driven one is more meaningful.

Reply:
IMO, since the long term goal is to have level driven controllers everywhere, it does seem right for the controller we add to follow that design.